cpu/atmega_common/periph/gpio.c

```diff
@@ -195,8 +301,80 @@ int gpio_init_int(gpio_t pin, gpio_mode_t mode, gpio_flank_t flank,
         return -1;
     }
 
-    /* not a valid interrupt pin */
+    /* not a valid interrupt pin. Set as pcint instead if pcints are enabled */
     if (int_num < 0) {
+        /* If pin change interrupts are enabled, enable mask and interrupt */
+ #ifdef PCINT_NUM_BANKS
+        int8_t offset = -1;
+        uint8_t pin_num = _pin_num(pin);
+
+        for (unsigned i = 0;
+             i < sizeof(pcint_mapping) / sizeof(pcint_mapping[0]); i++) {
+            if (pin != GPIO_UNDEF && pin == pcint_mapping[i]) {
+                offset = i;
+                break;
+            }
+        }
+
+        /* if pcint was not found: return -1  */
+        if (offset < 0) {
+            return offset;
+        }
+
+        unsigned bank = offset / 8;
+        DEBUG("PCINT enabled for bank %u offset %u\n", bank, (unsigned)offset);
+
+        /* save configuration for pin change interrupt */
+        pcint_config[offset].flank = flank;
+        pcint_config[offset].arg = arg;
+        pcint_config[offset].cb = cb;
+
+        /* init gpio */
+        gpio_init(pin, mode);
+        /* configure pcint */
+        cli();
+        switch (bank) {
+#ifdef PCINT0_IDX
+            case PCINT0_IDX:
+                PCMSK0 |= (1 << pin_num);
```

Comment:
Here again a bug: The pin number and the index in the PCINT bank do not match for the ATmega2560. I fixed the issue locally in https://github.com/RIOT-OS/RIOT/pull/11923/commits/8f669ae86f9e87d9a98a7094020a810c014e5433. With that everything works fine :-)

Reply:
Ok, cherry picked!